functions_context_this.js

```diff
@@ -1,3 +1,4 @@
+// calss replaced with function

```

Comment:
I would get rid of a condition:

```
if (fullNameAsArray.length === 2) {
        const [firstName, lastName] = fullNameAsArray;
        this.firstName = firstName;
        this.lastName = lastName;
 }
```
You can just take the 1st and the 2nd items.
Or if you want to leave this condition, I would propose to add an error for a consumer, since code is working silently, being not explicitly for a consumer.